src/Moq/MatcherFactory.cs

```diff
@@ -33,6 +35,23 @@ internal static class MatcherFactory
 
 		public static Pair<IMatcher, Expression> CreateMatcher(Expression argument, ParameterInfo parameter)
 		{
+			if (argument.NodeType == ExpressionType.Convert)
+			{
+				var convertExpression = (UnaryExpression)argument;
+				if (convertExpression.Method?.Name == "op_Implicit")
```

Comment:
Perhaps we could take `op_Implicit` handling even further and evaluate it if it isn't applied to an argument matcher.